### PR TITLE
add: import/export webwallet privateKey

### PR DESCRIPTION
**Pull request by hasankhadra**
Added functions importWebwallet and exportWebwallet to WebwalletContext. 
Can be used anywhere in the code.

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -298,6 +275,59 @@ function MyApp({ Component, pageProps }: AppPropsWithLayout) {
...
+	useEffect(() => {
+		if(!webwallet) {
+			return
+		}
+
+		if(nonce && nonce !== 'Token expired') {
+			return
+		}
+
+		addAuthorizedUser(webwallet?.address)
```
Comment (adiwajshing) [inline src/pages/_app.tsx:316]: Probably best to use async/await syntax 

Also I think the code needs to be linted once?

Comment (hasankhadra) [reply src/pages/_app.tsx]: How to use async/await syntax inside useEffect?
I pushed a batch. Let me know if that's what you meant.

File: src/pages/_app.tsx [hunk 2]
```diff
@@ -162,36 +164,11 @@ function MyApp({ Component, pageProps }: AppPropsWithLayout) {
 	// used to poll for scwAddress in "waitForScwAddress"
 	const scwAddressRef = useRef(scwAddress)
 
-	const getUseNonce = useCallback(async() => {
+	const getUseNonce = useCallback(async(): Promise<string> => {
```
Comment (adiwajshing) [inline src/pages/_app.tsx:167]: Why'd we need the extra type here?

File: src/pages/_app.tsx (hunk 1 above)
Comment (adiwajshing) [reply src/pages/_app.tsx]: async/await & promises are essentially the same thing, just differ in syntax. 

Eg of using this in a useEffect:
``` ts
useEffect(() => {
  (async() => {
    const abcd = await apiCall()
  })()
}, [])
```

File: src/pages/_app.tsx [hunk 3]
```diff
@@ -298,6 +275,59 @@ function MyApp({ Component, pageProps }: AppPropsWithLayout) {
...
+			await addAuthorizedUser(address)
+			return await getUseNonce()
+		}
+
+		getNonce(webwallet.address)
+		 .then((_nonce) => {
+				setNonce(_nonce)
+			})
+		 .catch((err) => {
+				logger.info({ err }, 'Error adding authorized user')
```
Comment (adiwajshing) [inline src/pages/_app.tsx:326]: shouldn't it be `logger.error`?

File: src/pages/_app.tsx (hunk 2 above)
Comment (hasankhadra) [reply src/pages/_app.tsx:167]: To have a better looking, cleaner code

Comment (adiwajshing) [reply src/pages/_app.tsx:167]: this is making things verbose I feel, Intellisense already tells us it's a string

File: src/pages/_app.tsx (hunk 3 above)
Comment (hasankhadra) [reply src/pages/_app.tsx:326]: Fixed.

File: src/pages/_app.tsx (hunk 1 above)
Comment (hasankhadra) [reply src/pages/_app.tsx]: I don't understand the difference between what I'm currently doing and what you proposed. Please elaborate more.

Comment (adiwajshing) [reply src/pages/_app.tsx]: you're using `then()` and `catch()` instead of the async function

Comment (hasankhadra) [reply src/pages/_app.tsx]: I'm using an async function and applying `.then()` and `.catch()` (which I need to set a state). 

```
useEffect(() => {
  (async() => {
    const abcd = await apiCall()
  })().then(() => {})
  .catch(() => {})
}, [])
```

This is your code with `.then()` and `.catch()`.

Comment (adiwajshing) [reply src/pages/_app.tsx]: You don't need then and catch for this, keeping it simple:
``` ts
useEffect(() => {
  (async() => {
    try {
      await addAuthorizedUser(address)
      const newNonce = await getUseNonce()
      setNonce(newNonce)
    } catch(err) {
      logger.error({err}, 'some error')
    }
  })()
}, [])

```

**Pull request merged**